src/Cosmonaut/CosmosStoreSettings.cs

```diff
@@ -76,5 +83,77 @@ public class CosmosStoreSettings
             DefaultCollectionThroughput = defaultCollectionThroughput;
             IndexingPolicy = indexingPolicy ?? CosmosConstants.DefaultIndexingPolicy;
         }
+
+        public void AddInterceptor<T>(Expression<Func<T, bool>> filter) where T : class
+        {
+            Interceptors.Add(new QueryInterceptor<T>(filter));
+        }
+    }
+
+    public interface IQueryInterceptor
+    {
+        Type Type { get; }
+    }
+
+    public class QueryInterceptor<TEntity> : ExpressionVisitor, IQueryInterceptor where TEntity : class
```

Comment:
Please move this, and it's interface into it's own class file.

Reply:
Resolved